cpp/locality/LinkCell.cc

```diff
@@ -128,8 +128,8 @@ const vec3<unsigned int> LinkCell::computeDimensions(const box::Box& box, float
     return dim;
 }
 
-bool compareFirstNeighborPairs(const std::vector<std::tuple<size_t, size_t, float>>& left,
-                               const std::vector<std::tuple<size_t, size_t, float>>& right)
+bool compareFirstNeighborPairs(const std::vector<std::tuple<size_t, size_t, float, float>>& left,
+                               const std::vector<std::tuple<size_t, size_t, float, float>>& right)
```

Comment:
I have a comment below about adding a `typedef` for bonds. It might actually be best to move that typedef to the `NeighborList`, and maybe have two separate typedefs, one with weights and one without (e.g. `typedef  std::tuple<size_t, size_t, float>` Bond; typedef std::tuple<size_t, size_t, float, float>` WeightedBond;`). I assume those will get unified at some point down the line.